src/modules/credentials/services/CredentialService.ts

```diff
@@ -249,26 +264,35 @@ export class CredentialService {
     connectionRecord.assertReady()
 
     // Create message
-    const { credentialDefinitionId, comment, preview } = credentialTemplate
+    const { credentialDefinitionId, comment, preview, linkedAttachments } = credentialTemplate
     const credOffer = await this.indyIssuerService.createCredentialOffer(credentialDefinitionId)
-    const attachment = new Attachment({
+    const offerAttachment = new Attachment({
       id: INDY_CREDENTIAL_OFFER_ATTACHMENT_ID,
       mimeType: 'application/json',
       data: new AttachmentData({
         base64: JsonEncoder.toBase64(credOffer),
       }),
     })
+
+    // Create and link credential to attacment
+    const credentialPreview = linkedAttachments
+      ? CredentialUtils.createAndLinkCredentialToAttachment(linkedAttachments, preview)
+      : preview
+
+    // Construct offer message
     const credentialOfferMessage = new OfferCredentialMessage({
       comment,
-      offerAttachments: [attachment],
-      credentialPreview: preview,
+      offerAttachments: [offerAttachment],
+      credentialPreview,
+      attachments: linkedAttachments?.map((linkedAttachment) => linkedAttachment.attachment),
```

Comment:
Just a thought: Instead of attachments, use `appendedAttachments`? This is how it is called in the Attachments RFC. https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0017-attachments/README.md#appending

However I'm not sure about this one. What do you think?

Reply:
Not the biggest fan. The method as discribed in the RFC is indeed appending, however every element in a list is appended in a way so the list already infers that it is using the appending method from the RFC.